### DISCUSSION TITLE
clocking time

### DISCUSSION THREAD

**Discussion by gour** [ad-si/TaskLite#98] [Q&A]
Hello,

would it be possible to somehow achieve clocking of tasks in Tasklite similar to taskwarrior+timewarrior?

**Comment by ad-si** [comment]
This is already supported! 
Although in a crude way and there are no features to analyze it yet.

```sh
$ tl add Practice singing
🆕 Added task "Practice singing" with id "01jq1042zbcy3bnfbk0d1dryy7"
$ tl start dryy7
🗒  Added a note to task "buy milk" with id "01jq1042zbcy3bnfbk0d1dryy7"
$ tl stop dryy7
🗒  Added a note to task "buy milk" with id "01jq1042zbcy3bnfbk0d1dryy7"
```

**Reply by gour** [reply]
> This is already supported! 

Hmm, that's interesting!

> Although in a crude way and there are no features to analyze it yet.

Do you have any plans in regard or it could be an opportunity to write external cli app to do it, e.g. in Haskell, as learning language experience ? :smile_cat:

**Reply by ogourment** [reply]
can you maybe provide sample use cases, eg how you used taskwarrior+timewarrior, to make it more concrete and create related issues where features are missing?

**Reply by apraga** [reply]
@ad-si Would it make sense to have a separate table for timetracking ? Storing the task id, start and end task for example. 
It would be easy to create reports from that but more code... The idea is to avoid running a query on all tasks, but maybe the performance cost is not that high ?
